src/System.Net.Http/src/System/Net/Http/Managed/ChunkedEncodingReadStream.cs

```diff
@@ -13,85 +14,81 @@ internal sealed partial class HttpConnection
+                _connection._allowedReadLineBytes = MaxChunkBytesAllowed;
+                ArraySegment<byte> line = await _connection.ReadNextLineAsync(cancellationToken).ConfigureAwait(false);
+                _connection._allowedReadLineBytes = currentAllowedReadLineBytes;
 
-            private ulong ParseHexSize(ArraySegment<byte> line)
-            {
-                if (line.Count == 0)
+                // Parse the hex value.
+                if (!Utf8Parser.TryParse(line.AsReadOnlySpan(), out ulong chunkSize, out int bytesConsumed, 'X'))
                 {
                     throw new IOException(SR.net_http_invalid_response);
                 }
-
-                ulong size = 0;
-                try
+                else if (bytesConsumed != line.Count)
                 {
-                    for (int i = 0; i < line.Count; i++)
+                    // There was data after the chunk size, presumably a "chunk extension".
+                    // Allow tabs and spaces and then stop validating once we get to an extension.
+                    int offset = line.Offset + bytesConsumed, end = line.Count - bytesConsumed + line.Offset;
+                    for (int i = offset; i < end; i++)
                     {
-                        char c = (char)line[i];
-                        if ((uint)(c - '0') <= '9' - '0')
+                        char c = (char)line.Array[i];
+                        if (c == ';')
                         {
-                            size = checked(size * 16 + ((ulong)c - '0'));
+                            break;
                         }
-                        else if ((uint)(c - 'a') <= ('f' - 'a'))
-                        {
-                            size = checked(size * 16 + ((ulong)c - 'a' + 10));
-                        }
-                        else if ((uint)(c - 'A') <= ('F' - 'A'))
-                        {
-                            size = checked(size * 16 + ((ulong)c - 'A' + 10));
-                        }
-                        else
+                        else if (c != ' ' && c != '\t') // not called out in the RFC, but WinHTTP allows it
                         {
                             throw new IOException(SR.net_http_invalid_response);
                         }
                     }
                 }
-                catch (OverflowException e)
+
+                _chunkBytesRemaining = chunkSize;
+                if (chunkSize > 0)
                 {
-                    throw new IOException(SR.net_http_invalid_response, e);
+                    return true;
                 }
-                return size;
+
+                // We received a chunk size of 0, which indicates end of response body. 
+                // Read and discard any trailing headers, until we see an empty line.
+                while (!LineIsEmpty(await _connection.ReadNextLineAsync(cancellationToken).ConfigureAwait(false)));
```

Comment:
As I said in another comment, I wonder if we care about accounting these things against MaxResponseHeaderLength.  We are just going to discard anyway.  Not having to maintain _allowedReadLineBytes across the body processing seems like it would simplify some logic.

Reply:
I agree it would simplify things a bit.  My only concern with that is if in the future we do start parsing and surfacing data from the trailing headers, at that point we will need a limit, and it seems like the existing limit would be a natural one to respect, but at that point it's a breaking change (though arguably a minor one).  Maybe it's so minor that it doesn't matter.  Ok.